app/queries/filter_rewards.rb

```diff
@@ -0,0 +1,17 @@
+class FilterRewards
+  attr_accessor :initial_scope
+
+  def initialize(initial_scope)
+    @initial_scope = initial_scope
+  end
+
+  def call(category)
```

Comment:
I suggest renaming it to results or scoped or something similar :) call rather belongs to Service Objects

```suggestion
  def call(category)
```